tests/unit_tests/test_make_pieces.py

```diff
@@ -21,15 +22,55 @@ def test_make_pieces(self, matrix, size, expected):
         actual = MakePiece.make_pieces(matrix, size)
         assert(actual == expected)
 
+    @pytest.mark.parametrize(
+        ("string, size,expected"),
+        [
+            # 空文字列
+            ("", 1, []),
+            ("", 3, []),
+            # 短めの文字列
+            ("aaabbbcccddd", 6, ["aaabbb", "cccddd"]),
+            ("aaabbbcccddd", 9, ["aaabbbccc", "ddd"]),
+            ("aaabbbcccddd", 12, ["aaabbbcccddd"]),
+            ("aaabbbcccddd", 15, ["aaabbbcccddd"]),
+            # 長めの文字列
+            ("".join(["a" for _ in range(1_000_000)]), 1,
+             ["".join(["a" for _ in range(1)]) for _ in range(1_000_000)]),
+            ("".join(["a" for _ in range(1_000_000)]), 1_000,
+             ["".join(["a" for _ in range(1_000)]) for _ in range(1_000)]),
+            ("".join(["a" for _ in range(1_000_000)]), 1_000_000,
+             ["".join(["a" for _ in range(1_000_000)]) for _ in range(1)]),
```

Comment:
```suggestion
            ('a'*1_000_000, 1,
             ['a'*1]*1_000_000),
            ('a'*1_000_000, 1_000,
             ['a'*1_000]*1_000),
            ('a'*1_000_000, 1_000_000,
             ['a'*1_000_000]*1),
```

Reply:
[IMO]
In terms of readability

Reply:
This is so beautiful. Thank you!
I did some rearranging and fixed it with 74c0a11424276159c651ef662c47d026a345291a